### ISSUE TITLE
Unable zoom in/out on Avalonia Android

### ISSUE THREAD

**Issue by vkotov** [Mapsui/Mapsui#2100]
**Mapsui Version**
4.0.0

**Mapsui Platform**
Avalonia 11.0.0

**Device**
Android

**Describe the bug**
Unable to zoom in map.

**To Reproduce**
1. Create Avalonia project using template (select android support).
2. Add <avalonia:MapControl Name="Map" /> to MainView.axaml
3. Add Map.Map?.Layers.Add(Mapsui.Tiling.OpenStreetMap.CreateTileLayer()); to MainView.axaml.cs (MainView constructor after InitializeComponent() call).
4. Run on Android device

**Expected behavior**
The map should be enlarged by double-tapping or swiping with two fingers.

**Screenshots**
![image](https://github.com/Mapsui/Mapsui/assets/1422494/0aaccfdc-0ff6-4bf4-bb77-5931ed5fab91)

**Comment by inforithmics** [comment]
Could you provide a simple Sample App to reproduce the behavior, Because currently There is only a Mapsui Avalnia Sample for Desktop.

But here I add more Sample targets for Avalonia
https://github.com/Mapsui/Mapsui/pull/2102

**Comment by vkotov** [comment]
> I have now a running Avalonia.Android Sample in #2102
Are you able to zoom in the map with your fingers?

**Comment by inforithmics** [comment]
I can confirm the behavior. It is because Multitouch gestures are not detected. From the look of the touched event it is the same as in mapsui.maui so the logic can be shared handling it.

**Comment by inforithmics** [comment]
Pull Request
https://github.com/Mapsui/Mapsui/pull/2127

**Comment by stevegdavis** [comment]
I still cannot get an Avalonia app to zoom, I can move the map OK but that's all I am using Mapsui Avalonia 4.0.0 and Avalonia 11.0.4 however the Mapsui 4.0.0 source samples do zoom OK?. 

My 2 sources MapView.axmal.cs

public partial class MainView : UserControl
{
    private MyLocationLayer? _myLocationLayer;
    public MainView()
    {
        InitializeComponent();
        var map = new Map();
        //mapView.Map.Layers.Add(OpenStreetMap.CreateTileLayer());
        mapView.Map.Navigator.RotationLock = false;

        mapView.UnSnapRotationDegrees = 30;
        mapView.ReSnapRotationDegrees = 5;


        _myLocationLayer?.Dispose();
        _myLocationLayer = new MyLocationLayer(map)
        {
            IsCentered = false,
        };

        map.Layers.Add(OpenStreetMap.CreateTileLayer());
        map.Layers.Add(_myLocationLayer);
        //map.Navigator.ZoomToLevel(14);
        var center = new MPoint(-2.218266, 51.745564);
        // OSM uses spherical mercator coordinates. So transform the lon lat coordinates to spherical mercator
        var sphericalMercatorCoordinate = SphericalMercator.FromLonLat(center.X, center.Y).ToMPoint();
        // Set the center of the viewport to the coordinate. The UI will refresh automatically
        // Additionally you might want to set the resolution, this could depend on your specific purpose
        map.Home = n => n.CenterOnAndZoomTo(sphericalMercatorCoordinate, n.Resolutions[9]);
        mapView.Map = map;
        _myLocationLayer.UpdateMyLocation(sphericalMercatorCoordinate, true);
    }
}
and in MainView.axmal

<mapsui:MapControl x:Name="mapView"/>

Android 11 screenshot:-

![Screenshot_20230929-141157](https://github.com/Mapsui/Mapsui/assets/15140454/a0bef988-b3e3-4243-bcd3-63799ce236bc)


I have tried different Avalonia templates but all are the same, mapsui.maui works fine. Apologies if it is something simple I am doing wrong

**Comment by slater1** [comment]
@stevegdavis Mapsui is working fine for me in Avalonia on Android 13. 

One thing I noticed is you are calling `new Map()` and assigning that to your `MapControl.Map` In my projeect, I tried that early on and also had problems. What you want to do is use the existing `MapControl.Map`. 

Example using MVVM:

```xaml
<!-- MapView.axaml --> 
    <maps:MapControl x:Name="MapControl" />
```

```csharp
// MapView.axaml.cs
  public MapView()
  {
    InitializeComponent();
    DataContextChanged += HandleDataContextChanged;
  } 

  private void HandleDataContextChanged(object? sender, EventArgs e)
  {
    if (DataContext is not IMapViewModel vm)
    {
      return;
    }

    vm.Map = MapControl;
  }

```

**Comment by slater1** [comment]
@stevegdavis you might need to use a CI build. There was a [merge](https://github.com/Mapsui/Mapsui/pull/2127) for touch in Avalonia on August 25. The latest release (4.0.0) is from June 17 and so doesn't have it. 

This what I am using

```csproj
  <PropertyGroup>
    <MapsuiVersion>4.0.0-173-g0ea83d675</MapsuiVersion>
  </PropertyGroup>
```

Available at https://github.com/Mapsui/Mapsui/actions/runs/5980554279. Scroll to bottom of page for the NuGet packages.

**Comment by stevegdavis** [comment]
@slater1 I had tried mapView.Map etc.. but that gave the same result. Will try other suggestions, many thanks for your reply. 
Also I noticed compared to Maui there is no Zoom buttons is showing properties, did you create your own Widgets to get zoom buttons? Thanks

**Comment by stevegdavis** [comment]
4.0.0-174-g2ba247d68 works, many thanks

**Comment by slater1** [comment]
Glad you got it.

My UI doesn't use any zoom buttons. I do have one custom button over the map, by placing both in the same control.

```axaml
    <maps:MapControl x:Name="MapControl" />
    <StackPanel Orientation="Horizontal" VerticalAlignment="Bottom">
      <CheckBox Content="Edit" IsChecked="{Binding Editing}" VerticalAlignment="Bottom"/>
    </StackPanel>
```

![image](https://github.com/Mapsui/Mapsui/assets/2089725/cc6cfc81-364b-4ef1-9e78-f1a27be6b255)

**Comment by pauldendulk** [comment]
@stevegdavis You can add the widget in code, like in the osm sample:

```csharp
namespace Mapsui.Samples.Common.Maps.Demo;

public class OsmSample : ISample
{
    public string Name => "1 OpenStreetMap";
    public string Category => "Demo";

    public Task<Map> CreateMapAsync()
    {
        return Task.FromResult(CreateMap());
    }

    public static Map CreateMap()
    {
        var map = new Map
        {
            CRS = "EPSG:3857"
        };
        map.Layers.Add(OpenStreetMap.CreateTileLayer());
        map.Widgets.Add(new ScaleBarWidget(map) { TextAlignment = Alignment.Center, HorizontalAlignment = HorizontalAlignment.Center, VerticalAlignment = VerticalAlignment.Top });
        map.Widgets.Add(new ZoomInOutWidget { MarginX = 20, MarginY = 40 });
        return map;
    }
}
```

**Comment by pauldendulk** [comment]
This should be implemented in 4.1.0-preview.1.
